crates/lib/src/cli.rs

```diff
@@ -733,6 +735,59 @@ pub(crate) fn require_root(is_container: bool) -> Result<()> {
     Ok(())
 }
 
+/// Check if a deployment can perform a soft reboot
+fn can_perform_soft_reboot(deployment: Option<&crate::spec::BootEntry>) -> bool {
+    deployment.map(|d| d.soft_reboot_capable).unwrap_or(false)
+}
+
+/// If there is staged deployment, check if soft reboot is possible and perform it if possible and return true
+/// else return false
+fn should_soft_reboot(
+    sysroot: &crate::store::Storage,
+    booted_deployment: Option<&ostree::Deployment>,
+) -> Result<bool> {
+    // Get updated status to check for soft-reboot capability
+    let (_deployments, updated_host) = crate::status::get_status(sysroot, booted_deployment)?;
+
+    // Check if there's a staged deployment that can perform soft reboot
+    if can_perform_soft_reboot(updated_host.status.staged.as_ref()) {
+        soft_reboot_staged(sysroot)?;
+        return Ok(true);
+    }
+    // TODO check if this reboot is doing a rollback...
+    // TODO then perform a soft reboot for the rollback deployment
+    Ok(false)
+}
+
+/// Prepare and execute a soft reboot for the given deployment
+#[context("Preparing soft reboot")]
+fn prepare_soft_reboot(
+    sysroot: &crate::store::Storage,
+    deployment: &ostree::Deployment,
+) -> Result<()> {
+    let cancellable = ostree::gio::Cancellable::NONE;
+    sysroot
+        .sysroot
+        .deployment_set_soft_reboot(deployment, false, cancellable)
```

Comment:
I know this is in draft, so my comment might be premature. But I think I'm missing where this executes the soft reboot. I see the `deployment_set_soft_reboot()` sets the ostree deployment:
https://github.com/ostreedev/ostree/blob/v2025.4/rust-bindings/src/auto/sysroot.rs#L178

Does this just setup the ostree deployment to be the next reboot target and all things required for the soft reboot, but not actually perform the soft reboot yet? Or am I missing something obvious? Probably the latter.